lib/src/crypto_subtle.dart

```diff
@@ -245,242 +300,375 @@ class Algorithm {
-  _getPropertyIfPresent(jsObj, 'dq', (String v) => jwk.dq = v);
-  _getPropertyIfPresent(jsObj, 'qi', (String v) => jwk.qi = v);
-  if (hasProperty(jsObj, 'oth')) {
-    final oth = getProperty(jsObj, 'oth');
-    final N = getProperty(oth, 'length');
-    jwk.oth = List.generate(N, (i) {
-      final jsObj = getProperty(oth, i);
-      return RsaOtherPrimesInfo(
-        r: getProperty(jsObj, 'r'),
-        d: getProperty(jsObj, 'd'),
-        t: getProperty(jsObj, 't'),
-      );
-    });
-  }
-  _getPropertyIfPresent(jsObj, 'k', (String v) => jwk.k = v);
-
-  return jwk;
+extension type JSJsonWebKey(JSObject _) implements JSObject {
+  external String? get kty;
+  external String? get use;
+  @JS('key_ops')
+  external JSArray<JSString>? get keyOps;
+  external String? get alg;
+  external bool? get ext;
+  external String? get crv;
+  external String? get x;
+  external String? get y;
+  external String? get d;
+  external String? get n;
+  external String? get e;
+  external String? get p;
+  external String? get q;
+  external String? get dp;
+  external String? get dq;
+  external String? get qi;
+  external JSArray<JSRsaOtherPrimesInfo>? get oth;
+  external String? get k;
 }
 
-/// Convert [JsonWebKey] to a javascript object.
-Object _jsonWebKeyToJsObj(JsonWebKey jwk) {
-  final jsObj = newObject();
-
-  _setPropertyIfPresent(jsObj, 'kty', jwk.kty);
-  _setPropertyIfPresent(jsObj, 'use', jwk.use);
-  _setPropertyIfPresent(jsObj, 'key_ops', jwk.key_ops);
-  _setPropertyIfPresent(jsObj, 'alg', jwk.alg);
-  _setPropertyIfPresent(jsObj, 'ext', jwk.ext);
-  _setPropertyIfPresent(jsObj, 'crv', jwk.crv);
-  _setPropertyIfPresent(jsObj, 'x', jwk.x);
-  _setPropertyIfPresent(jsObj, 'y', jwk.y);
-  _setPropertyIfPresent(jsObj, 'd', jwk.d);
-  _setPropertyIfPresent(jsObj, 'n', jwk.n);
-  _setPropertyIfPresent(jsObj, 'e', jwk.e);
-  _setPropertyIfPresent(jsObj, 'p', jwk.p);
-  _setPropertyIfPresent(jsObj, 'q', jwk.q);
-  _setPropertyIfPresent(jsObj, 'dp', jwk.dp);
-  _setPropertyIfPresent(jsObj, 'dq', jwk.dq);
-  _setPropertyIfPresent(jsObj, 'qi', jwk.qi);
-  final oth = jwk.oth;
-  if (oth != null) {
-    final jsArray = callConstructor(_array, [
-      oth.map((info) {
-        final jsObj = newObject();
-        setProperty(jsObj, 'r', info.r);
-        setProperty(jsObj, 'd', info.d);
-        setProperty(jsObj, 't', info.t);
-        return jsObj;
-      }).toList(),
-    ]);
-    setProperty(jsObj, 'oth', jsArray);
-  }
-  _setPropertyIfPresent(jsObj, 'k', jwk.k);
-
-  return jsObj;
+extension type JSRsaOtherPrimesInfo(JSObject _) implements JSObject {
+  external String get r;
+  external String get d;
+  external String get t;
 }
 
-/// Convert a [list] a Javascript object.
-Object _stringListToJsObj(List<String> list) => callConstructor(
-      _array,
-      list,
+TypedData getRandomValues(TypedData array) {
+  if (array is Uint8List) {
+    final result = window.crypto.getRandomValues(array.toJS) as JSUint8Array;
```

Comment:
We could perhaps make this simpler by throwing away the return value.

We can just `return array`, since that's also what the API returns:
https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues#return_value

Reply:
Thank you! 
I will give it a try!